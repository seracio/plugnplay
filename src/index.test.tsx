import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Plug, Playground, StoreContext } from './index';

test('Playground: only the child should be rendered', () => {
    const component = TestRenderer.create(
        <Playground store={{ test: 'test' }}>
            <div>hello world</div>
        </Playground>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Playground: the child component should have a context with a key "store"', () => {
    const App = () => {
        const store: any = React.useContext(StoreContext);
        return <div>{store.hello}</div>;
    };

    const component = TestRenderer.create(
        <Playground store={{ hello: 'hello world' }}>
            <App />
        </Playground>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Playground: props store should be mandatory', () => {
    const t = () => {
        TestRenderer.create(
            // @ts-ignore
            <Playground>
                <div>hello world</div>
            </Playground>
        );
    };
    expect(t).toThrow();
});
