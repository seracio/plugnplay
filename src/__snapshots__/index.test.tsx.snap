// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playground: only the child should be rendered 1`] = `
<div>
  hello world
</div>
`;

exports[`Playground: the child component should have a context with a key "store" 1`] = `
<div>
  hello world
</div>
`;

exports[`Plug: defaultValue should be used before stream is defined 1`] = `
<div>
  default value
</div>
`;

exports[`Plug: stream value should be used 1`] = `
<div>
  value
</div>
`;
